step/src/main/java/com/google/sps/CalendarManager.java

```diff
@@ -0,0 +1,150 @@
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.DateTime;
+import com.google.api.services.calendar.Calendar;
+import com.google.api.services.calendar.model.Event;
+import com.google.api.services.calendar.model.EventDateTime;
+import com.google.api.services.calendar.model.EventAttendee;
+import com.google.api.services.calendar.CalendarScopes;
+
+// TODO: method to return true/false is user is authenticated (javascript support too)
+// TODO: add javascript support for checking if a user is validated
+// TODO: add test cases
+// TODO: conform to Google Java style
+
+public class CalendarManager {
+  private CalendarManager() {}
+
+  public static void createMatchEvent(String hostUserId, String guestUserId,
+                                      int year, int month, int day, int hour, int minute) {
+    createMatchEvent(new User(hostUserId), new User(guestUserId),
+                      year, month, day, hour, minute);
+  }
+
+  public static void createMatchEvent(User hostUser, User guestUser,
+                                      int year, int month, int day, int hour, int minute) {
+    if (!hostUser.isAuthenticated()) {
+      throw new IllegalStateException("The host user isn't authenticated. "
+                                      + "Unable to create match event.");
+    }
+
+    Event matchEvent = new MatchEventBuilder()
+      .setAttendees(hostUser, guestUser)
+      .setStartDateTime(year, month, day, hour, minute)
+      .build();
+
+    pushMatchEvent(hostUser, guestUser, matchEvent);
+  }
+
+  public static List<String> getScopes() {
+    List<String> scopes = new ArrayList<>();
+    scopes.add(CalendarScopes.CALENDAR_EVENTS); // "View and edit events on all your calendars"
+    return scopes;
+  }
+
+  // hostUser is the event owner and guestUser will receive an email invite
+  private static void pushMatchEvent(User hostUser, User guestUser, Event event) {
+    Calendar calendar = getCalendar(hostUser);
+    try {
+      String calendarId = "primary";
+      event = calendar.events().insert(calendarId, event).execute();
+      System.out.printf("Event created: %s\n", event.getHtmlLink());
+    } catch (Exception e) {
+      System.out.println("Unable to push match event.");
+      e.printStackTrace();
+    }
+  }
+
+  private static Calendar getCalendar(User user) {
+    return new Calendar.Builder(
+        new NetHttpTransport(),
+        new JacksonFactory(),
+        user.getCredential()
+      ).setApplicationName("Friend Matching Plus").build(); // TODO: not sure if app name matters
+  }
+
+}
+
+class MatchEventBuilder {
+  private Event matchEvent;
+
+  public MatchEventBuilder() {
+    matchEvent = new Event();
+  }
+
+  public MatchEventBuilder setAttendees(User hostUser, User guestUser) {
+    matchEvent.setSummary("FMP: " + hostUser.getName() + " / " + guestUser.getName());
```

Comment:
Probably better to do `String.format("FMP: %s / %s", hostUser.getName(), guestUser.getName())` Makes it easier to understand the format.